### ISSUE TITLE
Operational Error (not being able to connect to the database)

### ISSUE THREAD

**Issue by Parvzae** [AllenInstitute/aisynphys#74]
I installed the database following the instruction in the website (Matrix analyzer). However, I cannot connect to the database. I get the following error:

  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/pyqtgraph/parametertree/parameterTypes.py", line 175, in valueChanged
self.widget.sigChanged.disconnect(self.widgetValueChanged)
TypeError: 'method' object is not connected

and when I press the update result bottom, I get this error:
sqlite3.OperationalError: no such table: patch_seq

The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "~/aisynphys/aisynphys/matrix_analyzer/matrix_analyzer.py", line 488, in update_clicked
    self.update_results()
  File "~/aisynphys/aisynphys/matrix_analyzer/matrix_analyzer.py", line 507, in update_results
    self.cell_groups, self.cell_classes = self.cell_class_filter.get_cell_groups(self.pairs)
  File "~/aisynphys/aisynphys/matrix_analyzer/matrix_analyzer.py", line 170, in get_cell_groups
    self.cell_groups = classify_cells(self.cell_classes, pairs=pairs)
  File "~/aisynphys/aisynphys/cell_class.py", line 209, in classify_cells
    if cell in cell_class:
  File "~/aisynphys/aisynphys/cell_class.py", line 89, in __contains

__
    patchseq = cell.patch_seq
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/orm/attributes.py", line 287, in __get__
    return self.impl.get(instance_state(instance), dict_)
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/orm/attributes.py", line 723, in get
    value = self.callable_(state, passive)
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/orm/strategies.py", line 759, in _load_for_state
    return self._emit_lazyload(
  File "<string>", line 1, in <lambda>
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/orm/strategies.py", line 900, in _emit_lazyload
    q(session)
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/ext/baked.py", line 545, in all
    return list(self)
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/ext/baked.py", line 445, in __iter__
    return q._execute_and_instances(context)
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3533, in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1011, in execute
    return meth(self, multiparams, params)
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 298, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1124, in _execute_clauseelement
    ret = self._execute_context(
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1316, in _execute_context
    self._handle_dbapi_exception(
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1510, in _handle_dbapi_exception
    util.raise_(
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "~/miniconda3/envs/aisynphys/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute
    cursor.execute(statement, parameters)

sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: patch_seq
[SQL: SELECT patch_seq.id AS patch_seq_id, patch_seq.cell_id AS patch_seq_cell_id, patch_seq.tube_id AS patch_seq_tube_id, patch_seq.nucleus AS patch_seq_nucleus, patch_seq.patchseq_hash AS patch_seq_patchseq_hash, patch_seq."result_BA" AS "patch_seq_result_BA", patch_seq.area_400_10000bp AS patch_seq_area_400_10000bp, patch_seq.picogreen_yield AS patch_seq_picogreen_yield, patch_seq.cluster_detail AS patch_seq_cluster_detail, patch_seq.cluster_label AS patch_seq_cluster_label, patch_seq.score AS patch_seq_score, patch_seq.res_index AS patch_seq_res_index, patch_seq.top_leaf AS patch_seq_top_leaf, patch_seq.top_leaf_score AS patch_seq_top_leaf_score, patch_seq.broad_class_label AS patch_seq_broad_class_label, patch_seq.subclass_label AS patch_seq_subclass_label, patch_seq.quality_score AS patch_seq_quality_score, patch_seq.norm_marker_sum AS patch_seq_norm_marker_sum, patch_seq.seurat_cluster AS patch_seq_seurat_cluster, patch_seq.seurat_score AS patch_seq_seurat_score, patch_seq.tree_first_cluster AS patch_seq_tree_first_cluster, patch_seq.tree_first_bt AS patch_seq_tree_first_bt, patch_seq.tree_first_kl AS patch_seq_tree_first_kl, patch_seq.tree_first_cor AS patch_seq_tree_first_cor, patch_seq.tree_second_cluster AS patch_seq_tree_second_cluster, patch_seq.tree_second_bt AS patch_seq_tree_second_bt, patch_seq.tree_second_kl AS patch_seq_tree_second_kl, patch_seq.tree_second_cor AS patch_seq_tree_second_cor, patch_seq.tree_third_cluster AS patch_seq_tree_third_cluster, patch_seq.tree_third_bt AS patch_seq_tree_third_bt, patch_seq.tree_call AS patch_seq_tree_call, patch_seq.genes_detected AS patch_seq_genes_detected, patch_seq.t_type AS patch_seq_t_type, patch_seq.meta AS patch_seq_meta 
FROM patch_seq 
WHERE ? = patch_seq.cell_id]
[parameters: (7263,)]
(Background on this error at: http://sqlalche.me/e/13/e3q8)

**Comment by scseeman** [comment]
Hi @Parvzae could you confirm that you are on the `current_release` branch?

**Comment by Parvzae** [comment]
Hi Stephanie,
I downloaded through
git clone https://github.com/alleninstitute/aisynphys
Are you recommending another method?
This is the version:
`python tools\matrix_analyzer.py --db-version=synphys_r1.0_2019-08-29_small.sqlite`

**Comment by scseeman** [comment]
I think that might put you on the `master` branch. If you do 
`git checkout current-release` 
that should put you on the proper branch for the tools to work. If that still doesn't solve the problem please let me know.

**Comment by Parvzae** [comment]
I checked it is already on the current-release branch. It says
`(base) 4120L-150444-M:aisynphys pzarei$ git checkout current-release
Already on 'current-release'
Your branch is up to date with 'origin/current-release'`
I still have the same problem.

**Comment by scseeman** [comment]
Hi @Parvzae, we'll take a look at this issue and get back to you as soon as we can. Sorry for the inconvenience!

**Comment by campagnola** [comment]
I just ran through the website instructions on a fresh system and wasn't able to reproduce the problem. However, I do remember running into this issue several months ago and fixing it here: https://github.com/AllenInstitute/aisynphys/commit/0b460661  (and indeed, if I revert those changes now then I get exactly the same error you have above). 

Is it possible that you are simply on an older or modified version of the `current-release` branch? You can check like this:

    $ git rev-parse HEAD
    eb262922604888dd200ff4b087e408d34d9da20b

    $ git status
    On branch current-release 
    Your branch is up to date with 'origin/current-release'.
    nothing to commit, working tree clean

If there are any differences, you may need to either pull the latest version of the code (`git pull origin current-release`) or fix any local modifications that may have been made.

**Comment by Parvzae** [comment]
Thank you, Luke!
I checked. I am on the current release branch. I even manually checked the file synphys_database.py. It is in the correct form. However, I am still getting the same error.
I tried on different computers. No success!

**Comment by campagnola** [comment]
I am stumped! Maybe if you post the output of `conda list --explicit`, we can try to reproduce this error?

**Comment by Parvzae** [comment]
Here is the output that I got:
```
conda list --explicit
# This file may be used to create an environment using:
# $ conda create --name <env> --file <this file>
# platform: osx-64
@EXPLICIT
https://repo.anaconda.com/pkgs/main/osx-64/blas-1.0-mkl.conda
https://repo.anaconda.com/pkgs/main/osx-64/ca-certificates-2020.7.22-0.conda
https://repo.anaconda.com/pkgs/main/osx-64/gettext-0.19.8.1-hb0f4f8b_2.conda
https://repo.anaconda.com/pkgs/main/osx-64/intel-openmp-2019.4-233.conda
https://repo.anaconda.com/pkgs/main/osx-64/jpeg-9b-he5867d9_2.conda
https://repo.anaconda.com/pkgs/main/osx-64/libcxx-10.0.0-1.conda
https://repo.anaconda.com/pkgs/main/osx-64/libgfortran-3.0.1-h93005f0_2.conda
https://repo.anaconda.com/pkgs/main/osx-64/libiconv-1.16-h1de35cc_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/llvm-openmp-10.0.0-h28b9765_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/xz-5.2.5-h1de35cc_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/yaml-0.2.5-haf1e3a3_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/zlib-1.2.11-h1de35cc_3.conda
https://repo.anaconda.com/pkgs/main/osx-64/expat-2.2.9-hb1e8313_2.conda
https://repo.anaconda.com/pkgs/main/osx-64/hdf5-1.10.6-hdbbcd12_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/icu-58.2-h0a44026_3.conda
https://repo.anaconda.com/pkgs/main/osx-64/libffi-3.3-hb1e8313_2.conda
https://repo.anaconda.com/pkgs/main/osx-64/libpng-1.6.37-ha441bb4_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/lz4-c-1.9.2-hb1e8313_1.conda
https://repo.anaconda.com/pkgs/main/osx-64/mkl-2019.4-233.conda
https://repo.anaconda.com/pkgs/main/osx-64/ncurses-6.2-h0a44026_1.conda
https://repo.anaconda.com/pkgs/main/osx-64/openssl-1.1.1h-haf1e3a3_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/pcre-8.44-hb1e8313_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/tk-8.6.10-hb0a8c7a_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/freetype-2.10.2-ha233b18_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/glib-2.65.0-hc5f4afa_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/libedit-3.1.20191231-h1de35cc_1.conda
https://repo.anaconda.com/pkgs/main/osx-64/readline-8.0-h1de35cc_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/zstd-1.4.5-h41d2c2f_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/dbus-1.13.16-h18a8e69_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/krb5-1.17.1-hddcf347_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/libtiff-4.1.0-hcb84e12_1.conda
https://repo.anaconda.com/pkgs/main/osx-64/sqlite-3.33.0-hffcf06c_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/lcms2-2.11-h92f6f08_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/libpq-12.2-h051b688_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/python-3.8.5-h26836e1_1.conda
https://repo.anaconda.com/pkgs/main/osx-64/qt-5.9.7-h468cd18_1.conda
https://repo.anaconda.com/pkgs/main/osx-64/certifi-2020.6.20-py38_0.conda
https://repo.anaconda.com/pkgs/main/noarch/cloudpickle-1.6.0-py_0.conda
https://repo.anaconda.com/pkgs/main/noarch/decorator-4.4.2-py_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/kiwisolver-1.2.0-py38h04f5b5a_0.conda
https://repo.anaconda.com/pkgs/main/noarch/olefile-0.46-py_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/psycopg2-2.8.5-py38hddc9c9b_0.conda
https://repo.anaconda.com/pkgs/main/noarch/pyparsing-2.4.7-py_0.conda
https://repo.anaconda.com/pkgs/main/noarch/pytz-2020.1-py_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/pyyaml-5.3.1-py38haf1e3a3_1.conda
https://repo.anaconda.com/pkgs/main/osx-64/sip-4.19.8-py38h0a44026_0.conda
https://repo.anaconda.com/pkgs/main/noarch/six-1.15.0-py_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/sqlalchemy-1.3.19-py38haf1e3a3_0.conda
https://repo.anaconda.com/pkgs/main/noarch/threadpoolctl-2.1.0-pyh5ca1d4c_0.conda
https://repo.anaconda.com/pkgs/main/noarch/toolz-0.11.1-py_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/tornado-6.0.4-py38h1de35cc_1.conda
https://repo.anaconda.com/pkgs/main/noarch/wheel-0.35.1-py_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/cycler-0.10.0-py38_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/cytoolz-0.11.0-py38haf1e3a3_0.conda
https://repo.anaconda.com/pkgs/main/noarch/dask-core-2.27.0-py_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/mkl-service-2.3.0-py38hfbe908c_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/pillow-7.2.0-py38ha54b6ba_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/pyqt-5.9.2-py38h655552a_2.conda
https://repo.anaconda.com/pkgs/main/noarch/python-dateutil-2.8.1-py_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/setuptools-49.6.0-py38_0.conda
https://repo.anaconda.com/pkgs/main/noarch/joblib-0.16.0-py_0.conda
https://repo.anaconda.com/pkgs/main/noarch/networkx-2.5-py_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/numpy-base-1.19.1-py38hcfb5961_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/pip-20.2.2-py38_0.conda
https://repo.anaconda.com/pkgs/main/noarch/imageio-2.9.0-py_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/h5py-2.10.0-py38h0601b69_1.conda
https://repo.anaconda.com/pkgs/main/osx-64/matplotlib-3.3.1-0.conda
https://repo.anaconda.com/pkgs/main/osx-64/matplotlib-base-3.3.1-py38h181983e_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/mkl_fft-1.2.0-py38hc64f4ea_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/mkl_random-1.1.1-py38h959d312_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/numpy-1.19.1-py38h3b9f5b6_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/pandas-1.1.1-py38hb1e8313_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/pywavelets-1.1.1-py38h1de35cc_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/scipy-1.5.0-py38hbab996c_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/patsy-0.5.1-py38_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/scikit-image-0.16.2-py38h6c726b0_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/scikit-learn-0.23.2-py38h959d312_0.conda
https://repo.anaconda.com/pkgs/main/osx-64/statsmodels-0.11.1-py38haf1e3a3_0.conda



```

**Comment by srarty** [comment]
Hi Luke,

I am having the exact same issue as @Parvzae . I made sure I am on the current-release branch and up to date. Below is my output to the `conda list --explicit` command.

It looks like we are both working from Australia. Do you think the issue could be related to some database access problem due to location or something similar?

```
conda list --explicit
# This file may be used to create an environment using:
# $ conda create --name <env> --file <this file>
# platform: win-64
@EXPLICIT
https://repo.anaconda.com/pkgs/main/win-64/blas-1.0-mkl.conda
https://repo.anaconda.com/pkgs/main/win-64/ca-certificates-2020.7.22-0.conda
https://repo.anaconda.com/pkgs/main/win-64/icc_rt-2019.0.0-h0cc432a_1.conda
https://repo.anaconda.com/pkgs/main/win-64/intel-openmp-2020.2-254.conda
https://repo.anaconda.com/pkgs/main/win-64/vs2015_runtime-14.16.27012-hf0eaf9b_3.conda
https://repo.anaconda.com/pkgs/main/win-64/mkl-2020.2-256.conda
https://repo.anaconda.com/pkgs/main/win-64/vc-14.1-h0510ff6_4.conda
https://repo.anaconda.com/pkgs/main/win-64/icu-58.2-ha925a31_3.conda
https://repo.anaconda.com/pkgs/main/win-64/jpeg-9b-hb83a4c4_2.conda
https://repo.anaconda.com/pkgs/main/win-64/lz4-c-1.9.2-hf4a77e7_3.conda
https://repo.anaconda.com/pkgs/main/win-64/openssl-1.1.1h-he774522_0.conda
https://repo.anaconda.com/pkgs/main/win-64/tk-8.6.10-he774522_0.conda
https://repo.anaconda.com/pkgs/main/win-64/xz-5.2.5-h62dcd97_0.conda
https://repo.anaconda.com/pkgs/main/win-64/yaml-0.2.5-he774522_0.conda
https://repo.anaconda.com/pkgs/main/win-64/zlib-1.2.11-h62dcd97_4.conda
https://repo.anaconda.com/pkgs/main/win-64/hdf5-1.10.4-h7ebc959_0.conda
https://repo.anaconda.com/pkgs/main/win-64/krb5-1.17.1-hc04afaa_0.conda
https://repo.anaconda.com/pkgs/main/win-64/libpng-1.6.37-h2a8f88b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/sqlite-3.33.0-h2a8f88b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/zstd-1.4.5-h04227a9_0.conda
https://repo.anaconda.com/pkgs/main/win-64/freetype-2.10.2-hd328e21_0.conda
https://repo.anaconda.com/pkgs/main/win-64/libpq-12.2-h3235a2c_0.conda
https://repo.anaconda.com/pkgs/main/win-64/libtiff-4.1.0-h56a325e_1.conda
https://repo.anaconda.com/pkgs/main/win-64/python-3.8.5-h5fd99cc_1.conda
https://repo.anaconda.com/pkgs/main/win-64/qt-5.9.7-vc14h73c81de_0.conda
https://repo.anaconda.com/pkgs/main/win-64/certifi-2020.6.20-py38_0.conda
https://repo.anaconda.com/pkgs/main/noarch/cloudpickle-1.6.0-py_0.conda
https://repo.anaconda.com/pkgs/main/noarch/decorator-4.4.2-py_0.conda
https://repo.anaconda.com/pkgs/main/win-64/kiwisolver-1.2.0-py38h74a9793_0.conda
https://repo.anaconda.com/pkgs/main/noarch/olefile-0.46-py_0.conda
https://repo.anaconda.com/pkgs/main/win-64/psycopg2-2.8.5-py38h7a1dbc1_0.conda
https://repo.anaconda.com/pkgs/main/noarch/pyparsing-2.4.7-py_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pyreadline-2.1-py38_1.conda
https://repo.anaconda.com/pkgs/main/noarch/pytz-2020.1-py_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pyyaml-5.3.1-py38he774522_1.conda
https://repo.anaconda.com/pkgs/main/win-64/sip-4.19.13-py38ha925a31_0.conda
https://repo.anaconda.com/pkgs/main/noarch/six-1.15.0-py_0.conda
https://repo.anaconda.com/pkgs/main/win-64/sqlalchemy-1.3.19-py38he774522_0.conda
https://repo.anaconda.com/pkgs/main/noarch/threadpoolctl-2.1.0-pyh5ca1d4c_0.conda
https://repo.anaconda.com/pkgs/main/noarch/toolz-0.11.1-py_0.conda
https://repo.anaconda.com/pkgs/main/win-64/tornado-6.0.4-py38he774522_1.conda
https://repo.anaconda.com/pkgs/main/noarch/wheel-0.35.1-py_0.conda
https://repo.anaconda.com/pkgs/main/win-64/wincertstore-0.2-py38_0.conda
https://repo.anaconda.com/pkgs/main/win-64/cycler-0.10.0-py38_0.conda
https://repo.anaconda.com/pkgs/main/win-64/cytoolz-0.11.0-py38he774522_0.conda
https://repo.anaconda.com/pkgs/main/noarch/dask-core-2.30.0-py_0.conda
https://repo.anaconda.com/pkgs/main/win-64/mkl-service-2.3.0-py38hb782905_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pillow-7.2.0-py38hcc1f983_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pyqt-5.9.2-py38ha925a31_4.conda
https://repo.anaconda.com/pkgs/main/noarch/python-dateutil-2.8.1-py_0.conda
https://repo.anaconda.com/pkgs/main/win-64/setuptools-50.3.0-py38h9490d1a_1.conda
https://repo.anaconda.com/pkgs/main/noarch/joblib-0.17.0-py_0.conda
https://repo.anaconda.com/pkgs/main/noarch/networkx-2.5-py_0.conda
https://repo.anaconda.com/pkgs/main/win-64/numpy-base-1.19.1-py38ha3acd2a_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pip-20.2.3-py38_0.conda
https://repo.anaconda.com/pkgs/main/noarch/imageio-2.9.0-py_0.conda
https://repo.anaconda.com/pkgs/main/win-64/h5py-2.10.0-py38h5e291fa_0.conda
https://repo.anaconda.com/pkgs/main/win-64/matplotlib-3.3.1-0.conda
https://repo.anaconda.com/pkgs/main/win-64/matplotlib-base-3.3.1-py38hba9282a_0.conda
https://repo.anaconda.com/pkgs/main/win-64/mkl_fft-1.2.0-py38h45dec08_0.conda
https://repo.anaconda.com/pkgs/main/win-64/mkl_random-1.1.1-py38h47e9c7a_0.conda
https://repo.anaconda.com/pkgs/main/win-64/numpy-1.19.1-py38h5510c5b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pandas-1.1.3-py38ha925a31_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pywavelets-1.1.1-py38he774522_2.conda
https://repo.anaconda.com/pkgs/main/win-64/scipy-1.5.0-py38h9439919_0.conda
https://repo.anaconda.com/pkgs/main/win-64/patsy-0.5.1-py38_0.conda
https://repo.anaconda.com/pkgs/main/win-64/scikit-image-0.16.2-py38h47e9c7a_0.conda
https://repo.anaconda.com/pkgs/main/win-64/scikit-learn-0.23.2-py38h47e9c7a_0.conda
https://repo.anaconda.com/pkgs/main/win-64/statsmodels-0.12.0-py38he774522_0.conda
```

**Comment by MWolfR** [comment]
Hello all,
I encountered the same error: 
`(sqlite3.OperationalError) no such table: patch_seq`

I tried asking the SynphysDatabase for its table_names and indeed: none named "path_seq":
```
In[7]: db.table_names()
Out[7]:
['avg_response_fit',
 'baseline',
 'baseline_response_strength',
 'cell',
 'dynamics',
 'electrode',
 'experiment',
 'gap_junction',
 'metadata',
 'morphology',
 'multi_patch_probe',
 'pair',
 'patch_clamp_recording',
 'pipeline',
 'pulse_response',
 'pulse_response_fit',
 'pulse_response_strength',
 'recording',
 'resting_state_fit',
 'slice',
 'stim_pulse',
 'stim_spike',
 'synapse',
 'synapse_prediction',
 'sync_rec',
 'test_pulse']
```

Curiously, when I try a different "db-version", the error changes:

with synphys_r2.0-pre3_small.sqlite, synphys_r2.0-pre2_small.sqlite and synphys_r2.0-pre1_small.sqlite 
the error is instead: 
`-- no such column: cell.is_excitatory`

Any ideas?

**Comment by MWolfR** [comment]
Okay, I managed to hack it to work.
Above someone posted a link to commit [0b460661](https://github.com/AllenInstitute/aisynphys/commit/0b460661), where a function is called to create missing table names. 

So I looked into that and found that - at least on my system - the function is never called!

I added in line 21 of matrix_analyzer a call to the function:
```
    print("Pre create tables:")
    print(db.table_names())
    db.create_tables()
    print("Post create tables:")
    print(db.table_names())
```
(Debug prints of course optional...)

And within "create_tables" I had to make another change. Line 499 from:
`self.ormbase.metadata.create_all(bind=self.rw_engine, tables=tables)`
to
`self.ormbase.metadata.create_all(bind=self.ro_engine, tables=tables)`

Because when running matrix_analyzer as documented then the rw_engine is None. At least for me.
